### DISCUSSION TITLE
Private Serverless Functions within Vercel Project

### DISCUSSION THREAD

**Discussion by n-glaz** [vercel/community#829] [Help] [answered]
I have an image caching CDN deployed via Vercel Serverless functions that serves transformed images via URL to another deployed app. 

I want to make sure that only requests from within the Vercel project (my deployed NextJS app in this case) can reach my Vercel serverless functions and their URLs. 

Is there documentation (or perhaps a guide if applicable) regarding restricting Vercel serverless function access?

**Comment by amyegan** [comment, marked as answer]
Thanks for asking @n-glaz! We do have a document describing some [authentication solutions](https://vercel.com/docs/concepts/solutions/authentication). More information is available in the [Next.js Authentication documentation](https://nextjs.org/docs/authentication). In addition to the examples linked on those pages, we also have a [Next.js JWT Authentication](https://vercel.com/templates/next.js/jwt-authentication) template.

Some others have shared their solutions for securing Serverless Functions here: https://github.com/vercel/vercel/discussions/5709
